pkg/controller/controller_instance.go

```diff
@@ -1226,6 +1248,48 @@ func (c *controller) prepareObservedGeneration(toUpdate *v1beta1.ServiceInstance
 		v1beta1.ServiceInstanceConditionFailed)
 }
 
+// updateServiceInstanceInProgressProperties updates the status with new
+// InProgressProperties value
+func (c *controller) updateServiceInstanceInProgressProperties(toUpdate *v1beta1.ServiceInstance, inProgressProperties *v1beta1.ServiceInstancePropertiesState) (*v1beta1.ServiceInstance, error) {
+	toUpdate.Status.InProgressProperties = inProgressProperties
+	updatedInstance, err := c.updateServiceInstanceStatus(toUpdate)
+	if err != nil {
+		return nil, err
+	}
+	return updatedInstance, nil
+}
+
+// isServiceInstancePropertiesStateEqual checks whether two ServiceInstancePropertiesState objects are equal
+func isServiceInstancePropertiesStateEqual(s1 *v1beta1.ServiceInstancePropertiesState, s2 *v1beta1.ServiceInstancePropertiesState) bool {
+	if s1 == nil && s2 == nil {
+		return true
+	}
+	if s1 == nil && s2 != nil || s1 != nil && s2 == nil {
```

Comment:
I know go will do the right thing, but to help readability, and newbie understandability, can you put parens around each side of the `&&` ?

Reply:
edited - sorry, I meant `||`